### ISSUE TITLE
Should recognize inputField when available in HTML

### ISSUE THREAD

**Issue by sguptatw** [getgauge/taiko#321]
**Expected behavior**
Should recognize inputField when available in HTML

**Actual behavior**
Getting `Input Field To Right of Element with text <text> not found` even if it is available in DOM

**Steps to replicate**
* Create HTML file 
```
<html>
<body>

<h2>The HTML Element</h2>
    <div>
        <button type="button" onclick="alert('Hello world!')">Click Me!</button>
        <input type="button" onclick="alert('Hello World!')" value="Click Me Also!">    
    </div>
    <div>
        First name: <input type="text" name="fname"><br>
        Last name: <input type="text" name="lname"><br>
        Birthday: <input type="date" name="bday">
        User password:<input type="password" name="psw">
        Select a file: <input type="file" name="myFile">
        <input type="submit" value="Submit">    
    </div>
    <div>
        <label for="male">Male</label>
        <input type="radio" name="gender" id="male" value="male"><br>
        <label for="female">Female</label>
        <input type="radio" name="gender" id="female" value="female"><br>
    </div>
    <div>
      <input type="checkbox" name="vehicle1" value="Bike"> I have a bike<br>
      <input type="checkbox" name="vehicle2" value="Car"> I have a car
    </div>
    <textarea></textarea>
</body>
</html>
```
* In the taiko REPL
```
> openBrowser()
 ✔ Browser opened
> goto("file:///.../HTMLElements.html")
 ✔ Navigated to url "file:///.../HTMLElements.html"
> write("First name",into(inputField(toRightOf(text("First name:")))))
 ✘ Error: Input Field To Right of Element with text "First name:" not found, run `.trace` for more info.
```

**Version**
```
commit - e3952e957e4c49f248c26b1fd9ce7505cc8b57a8
```

**Comment by NivedhaSenthil** [comment]
#### Theoretical idea for supporting proximity selectors for text nodes:

- Fetch text node containing the given text using `text()` in xpath (https://github.com/getgauge/taiko/blob/95d994bdccc4675d2242fe97db48521e8359c5f4/lib/taiko.js#L1854)
- Check visibility of textnode using parent node's offset (https://github.com/getgauge/taiko/blob/95d994bdccc4675d2242fe97db48521e8359c5f4/lib/taiko.js#L2298)
- use [range](https://stackoverflow.com/questions/16209153/how-to-get-the-position-and-size-of-a-html-text-node-using-javascript) to calculate position of the textnode in DOM handler if box model cant be calculated. (https://github.com/getgauge/taiko/blob/95d994bdccc4675d2242fe97db48521e8359c5f4/lib/domHandler.js#L80)
- then use it in proximity element search

**Comment by NivedhaSenthil** [comment]
Fix available in 7d43b16

**Comment by Debashis9012** [comment]
This issue has been tested and found fixed  with the version ``` 7d43b16 ```
